_posts/2016-07-11-value-of-automated-e2e-testing.md

```diff
@@ -0,0 +1,58 @@
+---
+layout: post
+author: riikkanen
+title: >
+  The Value of Automated e2e-testing in a software project
+excerpt: >
+  A peek into automated e2e-testing done in a software project. The first try with Wicket Page Tester. The second and more sophisticated version with JBehave and Selenium.
+tags:
+- Java
+- testing
+- e2e testing
+- JBehave
+- Wicket
+- Selenium
+- programming
+---
+
+The software project concerned started in the beginning of Autumn in 2010. I had the honor to work in the project from its beginning to June 2014 when my maternity leave began. The project was about to renew a 30 years old, technically outdated registry and everything attached to it, such as the user inteface and all the integrations. The project was a success, after almost three years of hard work, the application was taken into serious use without any major hardships. The project is still alive and kicking, even though, nowadays mostly maintenance, but also some feature development. 
+
+##The application
+
+The application is coded with Java. The actual user interface of the registry is developed as a web application using Apache Wicket, which is an Java web application framework. The data is stored in Oracle database and queried with direct SQL-queries, without any intermediate layer such as Hibernate. There are also numerous integrations bubbling under, powered by Mule. In the project I worked as a software developer both in back and front ends.
```

Comment:
"coded with Java" kuulostaa jotenkin hassulta ja finglishmiltä. The application is made with Java? The application is written in Java? En tiedä!
Voisko viimeinen lause olla myös jotain: "I worked as a full stack developer"? Sen saisi ehkä jotenkin ton ekan lauseen kanssa myös yhteen luontevasti, niin lähtee kappale käyntiin siististi :)